system/jlib/jptree.ipp

```diff
@@ -295,9 +386,18 @@ struct PtrStrUnion
         {
 #ifdef LITTLE_ENDIAN
             char flag;
-            char chars[sizeof(PTR *)-1];
-#else
-            char chars[sizeof(PTR *)-1];
+#endif
+            union
+            {
+                char chars[sizeof(PTR *)-1];
+                struct
+                {
+                    int8_t idx1;
+                    int16_t idx2;
+                    int32_t idx4;
+                };
```

Comment:
If 32bit , isn't this now too big?, i.e. int8+int16+in32 = 7 bytes + flag, so always 8 bytes and ptr 4.